engine/src/cmd/unit_generic.cpp

```diff
@@ -378,6 +375,12 @@ Unit::~Unit() {
 #ifdef DESTRUCTDEBUG
     VS_LOG_AND_FLUSH(trace, (boost::format("%1$d") % 0));
 #endif
+    for (size_t meshcount = 0; meshcount < meshdata.size(); ++meshcount) {
+        if (meshdata[meshcount] != nullptr) {
+            delete meshdata[meshcount];
+            meshdata[meshcount] = nullptr;
+        }
+    }
```

Comment:
@BenjamenMeyer This was a known issue with this version of the code (and an annoying one). This was the bug that I set out to fix in the branch that became my big refactor effort. Since we're reverting said branch, this bug is inevitably going to  return. At least for the time being.

I say let's go ahead and merge.